cpp/include/cugraph/detail/decompress_matrix_partition.cuh

```diff
@@ -180,6 +180,379 @@ void decompress_matrix_partition_to_fill_edgelist_majors(
+          thrust::sequence(thrust::seq,
+                           minor_map + major_offset,
+                           minor_map + major_offset + local_degree,
+                           adjacency_list_offset);
+        }
+      });
+  }
+  if (matrix_partition.get_dcs_nzd_vertex_count() &&
+      (*(matrix_partition.get_dcs_nzd_vertex_count()) > 0)) {
+    thrust::for_each(
+      execution_policy,
+      thrust::make_counting_iterator(vertex_t{0}),
+      thrust::make_counting_iterator(segment_offsets[4] - segment_offsets[3]),
+      [matrix_partition,
+       input_majors = input_majors + segment_offsets[3],
+       input_major_start_offsets =
+         input_major_start_offsets + segment_offsets[3] - segment_offsets[0],
+       majors,
+       minors,
+       property = property
+                    ? thrust::make_optional(thrust::make_tuple(
+                        thrust::get<0>(*property) + segment_offsets[3], thrust::get<1>(*property)))
+                    : thrust::nullopt,
+       global_edge_index] __device__(auto idx) {
+        auto major        = input_majors[idx];
+        auto major_offset = input_major_start_offsets[idx];
+        auto major_idx    = matrix_partition.get_major_hypersparse_idx_from_major_nocheck(major);
+        if (major_idx) {
+          vertex_t const* indices{nullptr};
+          thrust::optional<weight_t const*> weights{thrust::nullopt};
+          edge_t local_degree{};
+          thrust::tie(indices, weights, local_degree) =
+            matrix_partition.get_local_edges(*major_idx);
+          thrust::fill(
+            thrust::seq, majors + major_offset, majors + major_offset + local_degree, major);
+          thrust::copy(thrust::seq, indices, indices + local_degree, minors + major_offset);
+          if (property) {
+            auto major_input_property  = thrust::get<0>(*property)[idx];
+            auto minor_output_property = thrust::get<1>(*property);
+            thrust::fill(thrust::seq,
+                         minor_output_property + major_offset,
+                         minor_output_property + major_offset + local_degree,
+                         major_input_property);
+          }
+          if (global_edge_index) {
+            auto major_partition_offset =
+              static_cast<size_t>(*major_idx - matrix_partition.get_major_first());
+            auto adjacency_list_offset = thrust::get<0>(*global_edge_index)[major_partition_offset];
+            auto minor_map             = thrust::get<1>(*global_edge_index);
+            thrust::sequence(thrust::seq,
+                             minor_map + major_offset,
+                             minor_map + major_offset + local_degree,
+                             adjacency_list_offset);
+          }
+        }
+      });
+  }
+}
+
+template <typename vertex_t, typename edge_t, typename weight_t, typename prop_t, bool multi_gpu>
+void decompress_matrix_partition_to_fill_edgelist(
```

Comment:
Why do we have this function again? Or is this somewhat different from the one above?

Reply:
I changed the signature and removed the duplicate function in a later commit.